python/pyarrow/orc.py

```diff
@@ -175,3 +176,33 @@ def write_table(table, where):
     writer = ORCWriter(where)
     writer.write(table)
     writer.close()
+
+
+def read_table(source, columns=None, filesystem=None):
+    """
+    Read a table from ORC format
+
+    Parameters
+    ----------
+    source : str, pyarrow.NativeFile, or file-like object
+        If a string passed, can be a single file name or directory name. For
+        file-like objects, only read a single file. Use pyarrow.BufferReader to
+        read a file contained in a bytes or buffer-like object.
+    columns : list
+        If not None, only these columns will be read from the file. A column
+        name may be a prefix of a nested field, e.g. 'a' will select 'a.b',
+        'a.c', and 'a.d.e'. If empty, no columns will be read. Note
+        that the table will still have the correct num_rows set despite having
+        no columns.
```

Comment:
This last sentence doesn't actually seem to be true:

```
In [1]: import pyarrow as pa

In [2]: from pyarrow import orc

In [3]: orc.write_table(pa.table({'a': [1, 2, 3]}), "test.orc")

In [4]: result = orc.ORCFile("test.orc").read(columns=[])

In [5]: result.num_rows
Out[5]: 0

In [6]: result.num_columns
Out[6]: 0
```

Reply:
Thank you for pointing out that. Indeed. I didn't notice that.

Reply:
Since it needs a fix on `ORCFile` itself, can I handle it later in a separate PR, @jorisvandenbossche ?

Reply:
Oh, it seems that I can fix it in this layer.
```python
In [4]: result = orc.ORCFile("test.orc").read(columns=[])

In [5]: result.num_rows
Out[5]: 0

In [6]: result = orc.ORCFile("test.orc").read(columns=None)

In [7]: result.num_rows
Out[7]: 3

In [8]: result.num_columns
Out[8]: 1
```

Reply:
Hmm, it's maybe better to leave it for a separate JIRA, as reading in all columns to then select none of them in a second step might be a bit surprising for users?

Reply:
Thanks, @jorisvandenbossche . Yes, I agree with you that we need to optimize it in both `ORCFile` API and this one together for that case.